dsc/tests/dsc_args.tests.ps1

```diff
@@ -215,4 +215,60 @@ resources:
         $err.Length | Should -Not -Be 0
         $LASTEXITCODE | Should -Be 4
     }
+
+    It 'verify `dsc resource list` and `dsc resource list *`' {
+        # return all native resources, providers, but not adapter-based resources;
+        # results for `dsc resource list` and `dsc resource list *` should be the same
+        $a = dsc resource list -f json
+        $b = dsc resource list '*' -f json
+        $a.Count | Should -Be $b.Count
+        0..($a.Count-1) | %{
+            $a_obj = $a[$_] | ConvertFrom-Json
+            $b_obj = $b[$_] | ConvertFrom-Json
+            $a_obj.type | Should -Be $b_obj.type
+            # adapter-based resources should Not be in the results
+            $a_obj.requireAdapter | Should -BeNullOrEmpty
+            $b_obj.requireAdapter | Should -BeNullOrEmpty
+        }
+    }
+
+    It 'verify `dsc resource list resource_filter`' {
+        # same as previous but also apply resource_filter filter
+        $a = dsc resource list Test* -f json
```

Comment:
Similar to `*`, should put in single quotes so the shell doesn't try to interpret it and `dsc` receives it

```suggestion
        $a = dsc resource list 'Test*' -f json
```